barretenberg/cpp/src/barretenberg/solidity_helpers/circuits/recursive_circuit.hpp

```diff
@@ -89,52 +86,97 @@ template <typename OuterComposer> class RecursiveCircuit {
+            const auto& inputs = builder.public_inputs;
+            const auto recover_fq_from_public_inputs =
+                [&inputs, &builder](const size_t idx0, const size_t idx1, const size_t idx2, const size_t idx3) {
+                    const uint256_t l0 = builder.get_variable(inputs[idx0]);
+                    const uint256_t l1 = builder.get_variable(inputs[idx1]);
+                    const uint256_t l2 = builder.get_variable(inputs[idx2]);
+                    const uint256_t l3 = builder.get_variable(inputs[idx3]);
+
+                    const uint256_t limb = l0 + (l1 << NUM_LIMB_BITS_IN_FIELD_SIMULATION) +
+                                           (l2 << (NUM_LIMB_BITS_IN_FIELD_SIMULATION * 2)) +
+                                           (l3 << (NUM_LIMB_BITS_IN_FIELD_SIMULATION * 3));
+                    return outer_scalar_field(limb);
+                };
+
+            const auto x0 = recover_fq_from_public_inputs(builder.recursive_proof_public_input_indices[0],
+                                                          builder.recursive_proof_public_input_indices[1],
+                                                          builder.recursive_proof_public_input_indices[2],
+                                                          builder.recursive_proof_public_input_indices[3]);
+            const auto y0 = recover_fq_from_public_inputs(builder.recursive_proof_public_input_indices[4],
+                                                          builder.recursive_proof_public_input_indices[5],
+                                                          builder.recursive_proof_public_input_indices[6],
+                                                          builder.recursive_proof_public_input_indices[7]);
+            const auto x1 = recover_fq_from_public_inputs(builder.recursive_proof_public_input_indices[8],
+                                                          builder.recursive_proof_public_input_indices[9],
+                                                          builder.recursive_proof_public_input_indices[10],
+                                                          builder.recursive_proof_public_input_indices[11]);
+            const auto y1 = recover_fq_from_public_inputs(builder.recursive_proof_public_input_indices[12],
+                                                          builder.recursive_proof_public_input_indices[13],
+                                                          builder.recursive_proof_public_input_indices[14],
+                                                          builder.recursive_proof_public_input_indices[15]);
+            g1::affine_element P_affine[2]{
+                { x0, y0 },
+                { x1, y1 },
+            };
+
+            pairing_target_field result =
+                barretenberg::pairing::reduced_ate_pairing_batch_precomputed(P_affine, lines, 2);
+
+            return (result == pairing_target_field::one());
+        }
+        return true;
```

Comment:
To confirm; if builder.contains_recursive_proof and there are only 15 proof elements for example, this will return true -- is that what we want?

Reply:
I think this needs a separate issue - I copied this code from stdlib_recursion so it exists elsewhere

Reply:
actually I'll just fix. If it's not 16 that's a problem